pkg/planner/core/find_best_task.go

```diff
@@ -2623,7 +2628,11 @@ func (ds *DataSource) convertToPointGet(prop *property.PhysicalProperty, candida
 		found := false
 		for i := range ds.Columns {
 			if ds.Columns[i].ID == ds.HandleCols.GetCol(0).ID {
-				pointGetPlan.HandleColOffset = ds.Columns[i].Offset
+				if ds.HandleCols.GetCol(0).ID == model.ExtraHandleID {
+					pointGetPlan.HandleColOffset = -1
+				} else {
+					pointGetPlan.HandleColOffset = ds.Columns[i].Offset
```

Comment:
Should this also be applied to BatchPointGetPlan?

Reply:
dynamic prune + BatchPointGet is not allow right now, so I didn’t change it